### DISCUSSION TITLE
How to detected subscribes?

### DISCUSSION THREAD

**Discussion by Deku660** [pocketbase/pocketbase#4889] [Q&A]
I want to unsubscribe certain subscribe data, not the whole, but can I separate them in a way other than using "RECORD ID"?

**Comment by ganigeorgiev** [comment]
It is not clear what you are asking or trying to do. Please elaborate a little more on your use case and/or provide an example with the desired behavior.

**Reply by Deku660** [reply]
If I have the same "dm" table, but there are two subscribe codes with different filters, is there a way to separate them and unsubscribe? Except for Record Id, like the channel concept.

**Reply by ganigeorgiev** [reply]
I'm sorry, I still don't understand what you are trying to do.

If you are asking to filter the subscribed records you can provide an optional `filter` parameter to your subscribe call:

```js
// you can register multiple subscribe calls if you need to
const unsubscibeFunc = await pb.collection("example").subscribe("RECORD_ID", (e) => {
    ...
}, {
    filter: "someField != 'test'"
})
```

If that's not what you are trying to do, please elaborate more clearly on your use case to avoid the needless back-and-forth.

**Reply by Deku660** [reply]
Thanks.
but, i already using this filter.

A. 'example' table subscribe with filter,
B. 'example' table subscribe without filter.

If there are two codes that subscribe to one table like this, is there a way to unsubscribe only certain codes? A or B.
certain "RECORD_ID"

**Reply by ganigeorgiev** [reply]
I think we are going in circles and I won't be able to help you further as I still don't understand what you are trying to do (see [XY problem](https://xyproblem.info/)).

As a last attempt, in general you can call the individual `unsubscibeFunc` that is returned by each subscribe call. There is also a dedicated `pb.collection("example").unsubscribe([topic])` method but that will unsubscribe from all subscriptions for the specifed topic. For the available JS SDK methods you can check https://github.com/pocketbase/js-sdk#realtime-handlers.

**Reply by Deku660** [reply]
code A and B are subscribing same example table (their difference is only filter). How can I unsubscribe B?(except A). But exclude 'record_id'
